### PR TITLE
Simplify master-controller-manager

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This PR simplifies the master-controller-managers cmd with the goal of making the controllers that talk to seed clusters re-startable at runtime by a new controller which I will add in a follow-up.

Restarting controllers that talk to seeds at runtime when the seeds change is required for #2113 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @zreigz @nikhita

### CODE REVIEW

File: api/cmd/master-controller-manager/controllers.go [hunk 1]
```diff
@@ -1,180 +1,97 @@
...
+	}
+	if err := userprojectbinding.Add(ctrlCtx.mgr); err != nil {
+		return fmt.Errorf("failed to create userprojectbinding controller: %v", err)
+	}
+	if err := serviceaccount.Add(ctrlCtx.mgr); err != nil {
+		return fmt.Errorf("failed to create serviceaccount controller: %v", err)
 	}
-	return controllers, nil
+	if err := seedproxy.Add(ctrlCtx.mgr, 1, ctrlCtx.seedsGetter, ctrlCtx.seedKubeconfigGetter); err != nil {
+		return fmt.Errorf("failed to create seedproxy controllre: %v", err)
```
Comment (xrstf) [inline api/cmd/master-controller-manager/controllers.go]: *controller*

File: api/pkg/controller/rbac/rbac_controller_aggregator.go [hunk 2]
```diff
@@ -127,21 +128,26 @@ func New(metrics *Metrics, allClusterProviders []*ClusterProvider) (*ControllerA
...
+		workerCount:            workerCount,
 		rbacProjectController:  projectRBACCtrl,
 		rbacResourceController: resourcesRBACCtrl,
 		metrics:                metrics,
 	}, nil
 }
 
-// Run starts the controller's worker routines. This method is blocking and ends when stopCh gets closed
-func (a *ControllerAggregator) Run(workerCount int, stopCh <-chan struct{}) {
+// Run starts the controller's worker routines. Its an implementation of
```
Comment (xrstf) [inline api/pkg/controller/rbac/rbac_controller_aggregator.go]: *It's*

File: api/pkg/controller/rbac/rbac_controller_aggregator.go [hunk 3]
```diff
@@ -127,21 +128,26 @@ func New(metrics *Metrics, allClusterProviders []*ClusterProvider) (*ControllerA
...
 	defer runtime.HandleCrash()
 
-	go a.rbacProjectController.run(workerCount, stopCh)
-	go a.rbacResourceController.run(workerCount, stopCh)
+	go a.rbacProjectController.run(a.workerCount, stopCh)
+	go a.rbacResourceController.run(a.workerCount, stopCh)
 
 	glog.Info("RBAC generator aggregator controller started")
 	<-stopCh
+	glog.Infof("RBAC generator aggregator controller finished")
```
Comment (xrstf) [inline api/pkg/controller/rbac/rbac_controller_aggregator.go]: `.Info()`

**Comment by alvaroaleman** [comment]
@xrstf Updated, PTAL

**Pull request merged**